web/src/filter.js

```diff
@@ -135,16 +137,9 @@ function message_matches_search_term(message, operator, operand) {
             }
 
             // Try to match by stream_id if have a valid sub for
-            // the operand.
+            // the operand. If we can't find the id, we return false.
             const stream_id = stream_data.get_stream_id(operand);
-            if (stream_id) {
-                return message.stream_id === stream_id;
-            }
-
-            // We need this fallback logic in case we have a message
-            // loaded for a stream that we are no longer
-            // subscribed to (or that was deleted).
-            return message.stream.toLowerCase() === operand;
+            return stream_id && message.stream_id === stream_id;
```

Comment:
Hmm, this fallback issue might be interesting. 

I think there's nothing nice to be done here -- but we will get to eliminate the possibility that `stream_id` is invalid in this code path when we convert search to store what stream you're filtering to using pills, so it's can be `stream:invalid_stream_name`.

Reply:
Yeah, I think I incorporated the logic from the previous fallback issue into this code -- but instead of `false` being returned for different streams, we're working on if we can find a corresponding stream id at all.

Once we store stream_id in search, I agree that will be great :) There's still several places using `stream_name` because the search filter uses it.